igvc_navigation/src/path_planner/igvcsearchproblem.cpp

```diff
@@ -1,117 +1,190 @@
 #include "igvcsearchproblem.h"
 #include <math.h>
 
-SearchLocation robot_position;
 
-bool IGVCSearchProblem::isActionValid(SearchMove& move, SearchLocation start_state)
+// TODO update to include orientation when calculting CSpace
+// TODO what if I jump over an obstacle
+bool IGVCSearchProblem::isActionValid(const SearchMove& move, const SearchLocation& start_state)
 {
-  auto deltat = move.DeltaT;
-  double current = 0.0;
-  if (sqrt(pow(start_state.x - robot_position.x, 2) + pow(start_state.y - robot_position.y, 2)) > BoundingDistance)
-  {
-    return true;
-  }
-  while (current < (deltat + MaxObstacleDeltaT))
-  {
-    current = current > deltat ? deltat : (current + MaxObstacleDeltaT);
-    move.DeltaT = current;
-    SearchLocation result = getResult(start_state, move);
-    double offsetToCenter = 0.33;
-    pcl::PointXYZ searchPoint(result.x + offsetToCenter * cos(result.theta),
-                              result.y + offsetToCenter * sin(result.theta), 0);
-    std::vector<int> pointIdxRadiusSearch;
-    std::vector<float> pointRadiusSquaredDistance;
-    int neighborsCount = Octree->nearestKSearch(searchPoint, 1, pointIdxRadiusSearch, pointRadiusSquaredDistance);
-    if (neighborsCount > 0)
-    {
-      float distance = sqrt(pointRadiusSquaredDistance[0]);
-      if (distance < move.distToObs)
-      {
-        move.distToObs = distance;
-      }
-      if (move.distToObs <= Threshold)
-      {
-        return false;
-      }
-    }
+  SearchLocation result = getResult(start_state, move);
+  if (std::abs(result.ThetaChange) > MaxThetaChange)
+  {
+    return false;
+  }
+  double x = start_state.X + move.X;
+  double y = start_state.Y + move.Y;
+  if (x < 0 || y < 0 || x >= Map->image.size().width || y >= Map->image.size().height)
+  {
+    return false;
+  }
+  double min_val;
+  double max_val;
+  double sep = CSpace / Resolution;
+
+  // TODO how does this function worik
+  // implicit cast, TODO check a thing we want to do
```

Comment:
Did you turn these TODOs into TODONEs? If so, then you may want to add a comment or two describing what this does. If not, then can you explain what these are?

Reply:
@JasonGibson274